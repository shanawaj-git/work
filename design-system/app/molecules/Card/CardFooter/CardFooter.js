import React from 'react';
import PropTypes from 'prop-types';

import Typography, { TypographyType } from 'atoms/Typography';
import Button from 'atoms/Button';

import { TextWithIcon } from 'molecules/TextWithIcon/TextWithIcon';

export const CardFooter = props => {
  const { className, shouldShowButton = false, details = {} } = props;
  const { titleText, titleSideText, caption, buttonText } = details;
  return (
    <div className={`card-footer ${className}`}>
      <TextWithIcon
        text={titleText}
        sideText={titleSideText}
        className="card-footer-title"
      />
      <Typography
        typographyType={TypographyType.CAPTION}
        className="card-footer-subtitle"
      >
        {caption}
      </Typography>
      {shouldShowButton && (
        <Button variant="contained">
          <div>{buttonText}</div>
        </Button>
      )}
    </div>
  );
};

CardFooter.propTypes = {
  className: PropTypes.string,
  shouldShowButton: PropTypes.bool,
  details: PropTypes.shape({
    buttonText: PropTypes.string,
    titleText: PropTypes.string,
    titleSideText: PropTypes.string,
    caption: PropTypes.string,
  }),
};
