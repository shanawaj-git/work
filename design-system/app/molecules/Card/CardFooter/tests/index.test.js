import React from 'react';
import CardFooter from '..';

import renderer from 'react-test-renderer';

describe('index', () => {
  const details = {
    buttonText: 'buttonText',
    titleText: 'titleText',
    titleSideText: 'titleSideText',
    caption: 'caption',
  };
  it('should render card footer with no button', () => {
    const props = { className: 'test', shouldShowButton: false, details };

    const tree = renderer.create(<CardFooter {...props} />);
    expect(tree).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });

  it('should render card footer with button', () => {
    const props = { shouldShowButton: true, details };

    const tree = renderer.create(<CardFooter {...props} />);
    expect(tree).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });

  it('should render card footer with no details', () => {
    const props = {};

    const tree = renderer.create(<CardFooter {...props} />);
    expect(tree).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });
});
