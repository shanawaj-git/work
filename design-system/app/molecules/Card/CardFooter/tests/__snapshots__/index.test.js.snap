// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should render card footer with button 1`] = `
<div
  className="card-footer undefined"
>
  <div
    className="textWithIcon card-footer-title"
  >
    <span
      className="MuiTypography-root MuiTypography-caption css-1393mlh-MuiTypography-root"
    >
      titleText
    </span>
    <span
      className="MuiTypography-root MuiTypography-caption textWithIcon-sideText css-1393mlh-MuiTypography-root"
    >
      titleSideText
    </span>
  </div>
  <span
    className="MuiTypography-root MuiTypography-caption card-footer-subtitle css-1393mlh-MuiTypography-root"
  >
    caption
  </span>
  <button
    className="root StyledButton-sc-1apep4b-0 fcMYug MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      style={
        Object {
          "visibility": "visible",
        }
      }
    >
      <div>
        buttonText
      </div>
    </div>
  </button>
</div>
`;

exports[`index should render card footer with no button 1`] = `
<div
  className="card-footer test"
>
  <div
    className="textWithIcon card-footer-title"
  >
    <span
      className="MuiTypography-root MuiTypography-caption css-1393mlh-MuiTypography-root"
    >
      titleText
    </span>
    <span
      className="MuiTypography-root MuiTypography-caption textWithIcon-sideText css-1393mlh-MuiTypography-root"
    >
      titleSideText
    </span>
  </div>
  <span
    className="MuiTypography-root MuiTypography-caption card-footer-subtitle css-1393mlh-MuiTypography-root"
  >
    caption
  </span>
</div>
`;

exports[`index should render card footer with no details 1`] = `
<div
  className="card-footer undefined"
>
  <div
    className="textWithIcon card-footer-title"
  />
  <span
    className="MuiTypography-root MuiTypography-caption card-footer-subtitle css-1393mlh-MuiTypography-root"
  />
</div>
`;
